cid/helpers/quicksight/__init__.py

```diff
@@ -235,6 +235,15 @@ def create_data_source(self) -> bool:
                 current_status = datasource.status
             if not datasource.is_healthy:
                 logger.error(f'Data source creation failed: {datasource.error_info}')
+                if get_parameter(
+                    param_name='quicksight-delete-failed-datasource',
+                    message=f'Data source creation failed: {datasource.error_info}. Delete?',
+                    choices=['yes', 'no'],
+                ) == 'yes':
+                    try:
+                        self.delete_data_source(datasource.id)
+                    except self.client.exceptions.AccessDeniedException as e:
```

Comment:
this should be handled inside delete_data_source()

Reply:
ok

Reply:
well after some thoughts: how we can know from delete_datasource() if the error is critical or not. We know it only in the context of calling function. So error handling (or ignoring) must be also at this level.